driver-core/src/main/java/com/datastax/driver/core/TokenRange.java

```diff
@@ -166,6 +166,27 @@ public boolean intersects(TokenRange that) {
     }
 
     /**
+     * Returns whether this range contains the given range.
+     * <p/>
+     * For example:
+     * <ul>
+     * <li>{@code ]3,5]} contains {@code ]3,4]} and {@code ]4,5]};</li>
+     * <li>{@code ]3,5]} does not contain {@code ]1,2]}, {@code ]2,3]}, {@code ]5,7]}...</li>
+     * </ul>
+     *
+     * @param that the other range.
+     * @return whether this range contains the given range.
+     */
+    public boolean contains(TokenRange that) {
+        // Empty ranges never contain any other range
+        if (this.isEmpty() || that.isEmpty())
+            return false;
+
+        return this.contains(that.start, true)
+                && this.contains(that.end, false);
```

Comment:
This doesn't work if the other range has both ends inside but wraps around, for example `]4,8].contains(]7,5])` should be false.
A straightforward way to implement `contains` is `this.intersects(that) && this.intersectWith(that).equals(that)`. If we expose it, we can inline and maybe simplify the code.